cores/esp32/esp32-hal-uart.c

```diff
@@ -352,7 +353,7 @@ void uartSetBaudRate(uart_t* uart, uint32_t baud_rate)
         return;
     }
     UART_MUTEX_LOCK();
-    uint32_t clk_div = ((UART_CLK_FREQ<<4)/baud_rate);
+    uint32_t clk_div = ((rtc_clk_apb_freq_get()<<4)/baud_rate);
```

Comment:
There needs to be some limits between `APB` freq and `baud_rate`.  When I tested this at 4MHz, 3MHz (26MHz base clock) the serial monitor has garbage injected.
```
After change of clock to speed[7]=13
 rtc_clk_cpu_freq_get=1073470388, cpuFreq=13
 current apb freq =13000000, apbFreq=13
read 128 bytes

Error setting freq[8]=12MHZ


Error setting freq[9]=10MHZ


Error setting freq[10]=8MHZ


Error setting freq[11]=6MHZ

 After change of clock to speed[12]=5
 rtc_clk_cpu_freq_get=1073470388, cpuFreq=5
 current apb freq =5000000, apbFreq=5
read 127 bytes
⸮PW⸮Achange of clock to⸮.⸮⸮⸮m⸮3]=4
 rtc_clk⸮,⸮}⸮ɕ⸮}⸮⸮⸮⸮1073470388,⸮,⸮ɕ⸮⸮4
 current apb⸮⸮⸮⸮A=4000000, apbFreq=4
read 128⸮⸮ѕ⸮) After change of clock to⸮.⸮⸮⸮m⸮4]=3
⸮.⸮}⸮⸮⸮}⸮⸮⸮}⸮ɕ⸮}⸮⸮⸮⸮1073470388, cpuFreq=3
 current⸮X⸮Afreq =3000000, apbFreq=3
read⸮L&⸮bytes
 After change of clock to speed[15]=2
 rtc_clk_cpu_freq_get=4, cpuFreq=2
 current apb freq =2000000, apbFreq=2
read 127 bytes
 After change of clock to speed[16]=1
 rtc_clk_cpu_freq_get=1073470388, cpuFreq=1
 current apb freq =1000000, apbFreq=1
read 128 bytes
```
This is with Serial Monitor Baud set to 19200.

With Baud set to 115200 the ESP32 hangs when Frequency is changed to 1MHz. 
```
 After change of clock to speed[12]=5
 rtc_clk_cpu_freq_get=1073470388, cpuFreq=5
 current apb freq =5000000, apbFreq=5
read 127 bytes
⸮PW⸮Achange⸮⸮Aclock to speed[13]=4
⸮.⸮}⸮⸮⸮}⸮⸮⸮}⸮ɕ⸮}⸮⸮⸮⸮1073470388,⸮,⸮ɕ⸮⸮4
 current apb freq⸮O⸮⸮000, apbFreq=4
read 127⸮⸮ѕ⸮)⸮PW⸮Achange⸮⸮Aclock⸮⸮⸮⸮eed[14]=3
⸮.⸮}⸮⸮⸮}⸮⸮u⸮⸮⸮⸮}⸮⸮⸮⸮1073470388,⸮,⸮ɕ⸮⸮3
⸮⸮⸮ɕ⸮⸮Aapb⸮⸮⸮⸮A=3000000,⸮X⸮ɕ⸮⸮3
read⸮L⸮⸮ѕ⸮) After change of clock to speed[15]=2
 rtc_clk_cpu_freq_get=4, cpuFreq=2
 current apb freq =2000000, apbFreq=2
read 128 bytes
```
Testing code:
```c++
uint8_t speeds[] ={240,160,80,40,26,24,20,13,12,10,8,6,5,4,3,2,1};
  
void loop() {
  static int c=0;

  c++;
  if(c>=sizeof(speeds)) c = 0;
  
  Serial.flush();
  
  if(!setCpuFrequency(speeds[c])){
     Serial.printf("\nError setting freq[%d]=%dMHZ\n\n",c,speeds[c]);
  } else{
    delay(100);
    Serial.begin(BAUD);
    Wire.begin(SDA,SCL,100000);
    Serial.printf(" After change of clock to speed[%d]=%d\n",c,speeds[c]);
    Serial.printf(" rtc_clk_cpu_freq_get=%d, cpuFreq=%d\n",rtc_clk_cpu_freq_get(),getCpuFrequency());
    Serial.printf(" current apb freq =%d, apbFreq=%d\n",rtc_clk_apb_freq_get(),getApbFrequency());
    runBlocks();
    delay(5000);
    }
}
```
Chuck.

Reply:
I tested with 40MHz crystal :) and uart works fine. Thanks for testing 26MHz!